doc/compiler/runtime/Bootstrap.md

```diff
@@ -0,0 +1,113 @@
+<!--
+Copyright IBM Corp. and others 2025
+
+This program and the accompanying materials are made available under
+the terms of the Eclipse Public License 2.0 which accompanies this
+distribution and is available at https://www.eclipse.org/legal/epl-2.0/
+or the Apache License, Version 2.0 which accompanies this distribution and
+is available at https://www.apache.org/licenses/LICENSE-2.0.
+
+This Source Code may also be made available under the following
+Secondary Licenses when the conditions for such availability set
+forth in the Eclipse Public License, v. 2.0 are satisfied: GNU
+General Public License, version 2 with the GNU Classpath
+Exception [1] and GNU General Public License, version 2 with the
+OpenJDK Assembly Exception [2].
+
+[1] https://www.gnu.org/software/classpath/license.html
+[2] https://openjdk.org/legal/assembly-exception.html
+
+SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+-->
+
+# Background
+
+The OpenJ9 VM bootstraps in a staged approach. These stages are enumerated in
+the [`INIT_STAGE` `enum`](https://github.com/eclipse-openj9/openj9/blob/d4f97df2b8d823389467d4e9114c557b53a5d311/runtime/oti/jvminit.h#L196-L217).
+Among these stages, there are a few that are relevant to the JIT. This document
+describes the initialization that the JIT performs in each stage.
+
+The VM invokes the JIT hook
+```
+IDATA J9VMDllMain(J9JavaVM* vm, IDATA stage, void * reserved)
+```
+for each stage, passing the stage as an argument. The JIT then `switch`es on
+`stage`; in each `case` it does work appropriate for that stage.
+
+# Stages
+
+The following stages are in chronological order (in a JVM that starts up
+without error).
```

Comment:
Does the "stage" mean that the JIT is being called DURING that stage or once that stage is COMPLETE?

For example, what does `JIT_INITIALIZED` mean since the bulk of the initialization occurs during this stage?  Similarly, does `DLL_LOAD_TABLE_FINALIZED` mean the load table is finalized and can be relied upon by the JIT if necessary, or that the load table is not finalized until the end of this "stage" ?

Reply:
The JIT is called during the stage. In the `DLL_LOEAD_TABLE_FINALIZED` stage, once the VM has finished calling all components that have registered a `DLLMain` function, then I suppose the name of the stage will be true. 

I suppose in the case of `JIT_INITIALIZED`, it will be true once the JIT returns from the stage. However, I don't think there's a guarantee of the order in which components are hooked, so I don't think some other component can assume that if they're hooked in `JIT_INITIALIZED` that the JIT is actually "initialized"; it can depend on that once we go to the next stage.